cmd/dkimkeygen/main.go

```diff
@@ -1,54 +1,74 @@
 package main
 
 import (
+	"flag"
 	"fmt"
+	"github.com/driusan/dkim/pkg/algorithms"
 	"os"
 
-	"encoding/pem"
-	//	"encoding/gob"
-	//"encoding/asn1"
-	"crypto/x509"
 	"encoding/base64"
-
-	"crypto/rand"
-	"crypto/rsa"
+	"encoding/pem"
 )
 
 func main() {
-	pk, err := rsa.GenerateKey(rand.Reader, 2048)
-	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
+	var algorithmInput, outputPath, dnsOutputPath string
+	flag.StringVar(&algorithmInput, "a", "rsa-sha256", "Algorithm")
+	flag.StringVar(&outputPath, "o", "privkey.pem", "Save location for the PEM encoded private key")
+	flag.StringVar(&dnsOutputPath, "d", "dns.txt", "Save location for the TXT DNS entry")
+	flag.Parse()
+
+	if outputPath == "" {
+		_, _ = fmt.Fprintln(os.Stderr, "invalid output path")
 		os.Exit(1)
 	}
 
-	asn1bytes, err := x509.MarshalPKIXPublicKey(&pk.PublicKey)
+	if dnsOutputPath == "" {
+		_, _ = fmt.Fprintln(os.Stderr, "invalid DNS output path")
```

Comment:
I guess [this is worth a read](https://travix.io/unhandled-errors-in-go-3f341f2704dd), but yes, I do partially agree on the `log.Fatal`, the problem with that is that it will remove entirely all of the other exit codes:

```
 Fatalf is equivalent to Printf() followed by a call to os.Exit(1). 
```